src/bin/hexyl.rs

```diff
@@ -179,23 +182,50 @@ fn main() {
     }
 }
 
-fn parse_byte_count(n: &str, block_size: u64) -> Option<u64> {
+#[derive(Clone, Copy, Debug, Default, Hash, Eq, Ord, PartialEq, PartialOrd)]
+pub struct PositiveI64(i64);
+
+impl PositiveI64 {
+    pub fn new(x: i64) -> Option<Self> {
+        if x.is_negative() {
+            None
+        } else {
+            Some(Self(x))
+        }
+    }
+
+    pub fn into_inner(self) -> i64 {
+        let Self(x) = self;
+        x
+    }
+}
+
+impl Into<u64> for PositiveI64 {
+    fn into(self) -> u64 {
+        let Self(x) = self;
+        u64::try_from(x).unwrap()
```

Comment:
Minor: I like to use `.expect(…)` instead of `.unwrap()`, if only for documentation purposes. Something like:
```rust
try_from(x).expect("Inner value is guaranteed to be positive")
```

Reply:
:+1:

Reply:
Updated -- how's it look?